dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+  }
+}
+
+class SvgPathCommandBuilder {
+  static const Map<String, Null> kRelativeCommands = const <String, Null> {
+    'c': null,
+    'l': null,
+    'm': null,
+    't': null,
+    's': null
+  };
+
+  Point<double> lastPoint = const Point<double>(0.0, 0.0);
+
+  SvgPathCommand build(String type, List<Point<double>> points) {
+    List<Point<double>> absPoints = points;
+    if (_isRelativeCommand(type)) {
+      absPoints = points.map((Point<double> p) => p + lastPoint).toList();
+    }
+
+    // closePath commands are always last in the path, so we don't need to track
+    // last point.
+    if (type != 'Z' && type != 'z')
+      lastPoint = absPoints.last;
+
+    return new SvgPathCommand(type.toUpperCase(), absPoints);
+  }
+
+  static bool _isRelativeCommand(String type) {
+    return kRelativeCommands.containsKey(type);
+  }
+}
+
+List<double> _pointsToVector3Array(List<Point<double>> points) {
+  final List<double> result = new List<double>(points.length * 3);
+  for (int i = 0; i < points.length; i += 1) {
+    result[i * 3] = points[i].x;
+    result[i * 3 + 1] = points[i].y;
+    result[i * 3 + 2] = 1.0;
+  }
+  return result;
+}
+
+List<Point<double>> _vector3ArrayToPoints(List<double> vector) {
+  final int numPoints = (vector.length / 3).floor();
+  final List<Point<double>> points = new List<Point<double>>(numPoints);
+  for (int i = 0; i < numPoints; i += 1) {
+    points[i] = new Point<double>(vector[i*3], vector[i*3 + 1]);
+  }
+  return points;
+}
+
+/// Represents a transformation to apply on an SVG subtree.
+///
+/// This includes more transforms than the ones described by the SVG transform
+/// attribute, e.g opacity.
+class _Transform {
+
+  /// Constructs a new _Transform, default arguments create a no-op transform.
+  _Transform({Matrix3 transformMatrix, this.opacity = 1.0}) :
+      this.transformMatrix = transformMatrix ?? new Matrix3.identity();
+
+  final Matrix3 transformMatrix;
+  final double opacity;
+
+  _Transform applyTransform(_Transform transform) {
+    return new _Transform(
+        transformMatrix: transform.transformMatrix.multiplied(transformMatrix),
+        opacity: transform.opacity * opacity,
+    );
+  }
+}
+
+final RegExp _transformCommand = new RegExp(' *(translate|scale|rotate)\\(([^)]*)\\)');
+
+Matrix3 _parseSvgTransform(String transform){
+  final Iterable<Match> matches =_transformCommand.allMatches(transform).toList().reversed;
+  Matrix3 result = new Matrix3.identity();
+  for (Match m in matches) {
+    final String command = m.group(1);
+    final String params = m.group(2);
+    if (command == 'translate') {
+      result = _parseSvgTranslate(params).multiplied(result);
+      continue;
+    }
+    if (command == 'scale') {
+      result = _parseSvgScale(params).multiplied(result);
+      continue;
+    }
+    if (command == 'rotate') {
+      result = _parseSvgRotate(params).multiplied(result);
+      continue;
+    }
+    throw new Exception('unimplemented transform: $command');
```

Comment:
you can't hit this given your pattern.

As before with the paths, I think we should implement this by walking the string, rather than only looking at matches.

Reply:
made the regex less tight